src/tensors/cpu/wasm_intgemm_interface.h

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+/* Main interface for integer matrix multiplication (C = A * B) for wasm.
+ *
+ * A is typically activations whose rows should be a multiple of 1 (i.e. no restriction) and
+ * columns should be a multiple of 64.
+ * 
+ * B is typically fixed model parameters whose rows should be a multiple of 64 and columns
+ * should be a multiple of 8.
+ * 
+ * C is row major.
+ */
+
+#include <cstdint>
+
+using Index = unsigned int;
```

Comment:
I will change `Index` to a fixed-width data type `unit32_t`.

Reply:
We use `unsigned int` https://github.com/kpu/intgemm/blob/master/intgemm/types.h#L56 . We may need to negotiate this change before you change it unliterary

Reply:
@kpu @XapaJIaMnu Could there be any issue if I change it to `uint32_t`? It would be great if we can adopt a fixed-width data type in these APIs.

Reply:
64-bit support is a good thing to have in the general library but I would be ok with an option.

Reply:
I *think* not, but we should be consistent when using those datatypes.

Reply:
This interface is for FFI purpose only, original intgemm code mostly shall stay unaffected. My recommendation is to define different type e.g. `IndexForFFI` define it as e.g. `uint32_t`, use in this interface instead of `Index`, and implementation assert this assumption in the library?

Reply:
@kpu @XapaJIaMnu  Should I go ahead with `uint32_t` then? We can use assert for `sizeof(Index) == sizeof(IndexForFFI)` as @yurydelendik suggested.

Reply:
Is there are problem if we use `uint64_t` since this is what *should* be used to address 64 bit arrays. Otherwise we may run into issue with (very) big matrices.

Reply:
to be flexible `size_t` has to be use, which reflect addressable pointer size

Reply:
Defining `Index` as `uint64_t` will cripple performance on 32-bit platforms. Defining as `unsigned int` is on 64-bit and 32-bit is mostly, but not guaranteed, is 32 bit. `uint32_t` just makes it official.

Reply:
`size_t` is `4-bytes` for wasm platform and I had encountered a bug in marian because `size_t` and `uint64_t` were being used interchangeably.

However, `Index` parameter is being used to refer to the indices of an array and from that logic it should be `size_t`.

What is the final call on this? Can the `Index` ever be bigger than the max value of `uint32_t`?

Reply:
My recommendation to set it to uint32_t in this header. (In the implementation of the function you can just assert `sizeof(Index) == sizeof(uint32_t)`. Only 16-bit platforms will be affected by explicitly making `uint32_t` here.

Reply:
Changing `Index` to `uint32_t` as
1. It is set to `unsigned int` in intgemm's existing interface
2. The number of model parameters are hardly going to touch such a high number

Please let me know if anyone has a concern. I am making the change.